NeuroID/DataStore.swift

```diff
@@ -28,6 +29,20 @@ public enum DataStore {
     }
 
     static func cleanAndStoreEvent(screen: String, event: NIDEvent, storeType: String) {
+        
+        // If queue has more than 2000 events, send a queue full event and return
+        if (DataStore.queuedEvents.count >= max_event_size || DataStore.events.count >= max_event_size) {
+            if (DataStore.events.last?.type != NIDEventName.bufferFull.rawValue) {
+                var fullEvent = NIDEvent.init(type: NIDEventName.bufferFull)
+                if storeType == "queue" {
+                    DataStore.queuedEvents.append(fullEvent)
+                } else {
+                    DataStore.events.append(fullEvent)
+                }
+            }
+            return
```

Comment:
One other note, we should log a warning here like in android that we have a full buffer and are dropping X event type